test/app/commands.test.ts

```diff
@@ -0,0 +1,339 @@
+    let settingsManagerStub: sinon.SinonStubbedInstance<
+      models.ISettingsManager
+    >;
+    let notifyManagerStub: sinon.SinonStubbedInstance<
+      models.INotificationManager
+    >;
+    let iconsGeneratorStub: sinon.SinonStubbedInstance<models.IIconsGenerator>;
+    let padMngStub: sinon.SinonStubbedInstance<
+      models.IProjectAutoDetectionManager
+    >;
+    let onDidChangeConfigurationStub: sinon.SinonStub;
+    let registerCommandStub: sinon.SinonStub;
+    let showCustomizationMessageStub: sinon.SinonStub;
+    let togglePresetStub: sinon.SinonStub;
+    let extensionManager: models.IExtensionManager;
+
+    beforeEach(function () {
+      sandbox = sinon.createSandbox();
+
+      vscodeManagerStub = sandbox.createStubInstance<models.IVSCodeManager>(
+        VSCodeManager
+      );
+      sandbox.stub(vscodeManagerStub, 'context').get(() => ({
+        subscriptions: [],
+      }));
+      onDidChangeConfigurationStub = sandbox.stub();
+      sandbox.stub(vscodeManagerStub, 'workspace').get(() => ({
+        onDidChangeConfiguration: onDidChangeConfigurationStub,
+      }));
+      registerCommandStub = sandbox.stub();
+      sandbox.stub(vscodeManagerStub, 'commands').get(() => ({
+        registerCommand: registerCommandStub,
+        executeCommand: sandbox.stub(),
+      }));
+
+      configManagerStub = sandbox.createStubInstance<models.IConfigManager>(
+        ConfigManager
+      );
+
+      settingsManagerStub = sandbox.createStubInstance<models.ISettingsManager>(
+        SettingsManager
+      );
+      notifyManagerStub = sandbox.createStubInstance<
+        models.INotificationManager
+      >(NotificationManager);
+
+      iconsGeneratorStub = sandbox.createStubInstance<models.IIconsGenerator>(
+        IconsGenerator
+      );
+
+      padMngStub = sandbox.createStubInstance<
+        models.IProjectAutoDetectionManager
+      >(ProjectAutoDetectionManager);
+
+      extensionManager = new ExtensionManager(
+        vscodeManagerStub,
+        configManagerStub,
+        settingsManagerStub,
+        notifyManagerStub,
+        iconsGeneratorStub,
+        padMngStub
+      );
+
+      showCustomizationMessageStub = sandbox.stub(
+        extensionManager,
+        // @ts-ignore
+        'showCustomizationMessage'
+      );
+
+      togglePresetStub = sandbox.stub(
+        extensionManager,
+        // @ts-ignore
+        'togglePreset'
+      );
+    });
+
+    afterEach(function () {
+      sandbox.restore();
+    });
+
+    context(`registering the commands`, function () {
+      let commands: any[];
+      let reflectGetSpy: sinon.SinonSpy;
+
+      beforeEach(function () {
+        reflectGetSpy = sandbox.spy(Reflect, 'get');
+      });
+
+      it(`reflects the provided callbacks`, function () {
+        commands = [
+          { command: 'activateIcons', callbackName: 'activationCommand' },
+        ];
+
+        // @ts-ignore
+        extensionManager.registerCommands(commands);
```

Comment:
Private functions shouldn't be tested directly but using the public interfaces. I understand you did this to make things easier but I think we should avoid this kind of practices and try to fix it in next iterations.

Reply:
I was expecting you would say something like this. I'm aware. My intention was to keep the public API to the minimum necessary. Unfortunately, this had that effect on the testing cases. Personally, I don't think it's a big deal since it's only a `TS` rule. I tried to do it the way you suggest and it got messy pretty quick.

Reply:
As I said I totally understand your decision. It's not something critical, so never mind. We can try to refactor this any time further if we feel like it. I just wanted to point this out so we don't make of this a rule of thumb for our test cases.